docs/source/model/dependencies.rst

```diff
@@ -0,0 +1,453 @@
+        )
+
+    # => This will work
+
+.. warning::
+
+    By the same reason, ``code_path`` option doesn't handle the relative import like ``code_path=["../src"]``.
+
+
+.. _validating-environment-for-prediction:
+
+Validating Environment for Prediction
+-------------------------------------
+
+Validating your model before deployment is a critical step to ensure production readiness.
+MLflow provides a few ways to test your model locally, either in a virtual environment or a Docker container.
+If you find any dependency issues during validation, please follow the guidance in :ref:`How to fix dependency errors when serving my model? <how-to-fix-dependency-errors-in-model>`
+
+Testing offline prediction with a virtual environment
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+You can use MLflow **predict** API via CLI or Python to make test predictions with your model.
+This will load your model from the model URI, create a virtual environment with the model dependencies (defined in MLflow Model),
+and run offline predictions with the model.
+Please refer to :py:func:`mlflow.models.predict` or `CLI reference <../cli.html#mlflow-models>`_ for more detailed usage for the predict API.
+
+.. tabs::
+
+    .. code-tab:: bash
+
+        mlflow models predict -m runs:/<run_id>/model-i <input_path>
+
+    .. code-tab:: python
+
+        import mlflow
+
+        mlflow.models.predict(
+            model_uri="runs:/<run_id>/model",
+            input_data=<input_data>,
+        )
+
+Using the ``mlflow models predict`` API is convenient for testing your model and inference environment quickly.
+However, it may not be a perfect simulation of the serving because it does not start the online inference server.
+
+Testing online inference endpoint with a virtual environment
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+If you want to test your model by actually running the online inference server, you can use MLflow `serve` API.
+This will create a virtual environment with your model and dependencies, similarly to the `predict` API, but will start the inference server
+and expose the REST endpoints. Then you can send a test request and validate the response.
+Please refer to `CLI reference <../cli.html#mlflow-models>`_ for more detailed usage for the `serve` API.
+
+.. code-block:: bash
+
+    mlflow models serve -m runs:/<run_id>/model -p <port>
+    # In another terminal
+    curl -X POST -H "Content-Type: application/json" \
+        --data '{"inputs": [[1, 2], [3, 4]]}' \
+        http://localhost:<port>/invocations
+
+While this is a reliable way to test your model before deployment, one caveat is that the virtual environment doesn't absorb the OS-level differences
+between your machine and the production environment. For example, if you are using MacOS as a local dev machine but your deployment target is
+running on Linux, you may encounter some issues that are not reproducible in the virtual environment.
```

Comment:
```suggestion
running on a Linux virtual environment, you may encounter some issues that are not reproducible in the virtual environment.
```

Reply:
this is not necessarily a virtual environment - like EC2, k8s cluster on on-premise Linux machines, etc:)

Reply:
ah, good point!